gtdynamics/factors/PoseFactor.h

```diff
@@ -59,17 +59,16 @@ inline gtsam::NoiseModelFactor::shared_ptr PoseFactor(
  *
  * @param wTp_key Key for parent link's CoM pose in world frame.
  * @param wTc_key Key for child link's CoM pose in world frame.
- * @param q_key Key for joint value.
  * @param cost_model The noise model for this factor.
  * @param joint The joint connecting the two poses
  */
 inline gtsam::NoiseModelFactor::shared_ptr PoseFactor(
-    DynamicsSymbol wTp_key, DynamicsSymbol wTc_key, DynamicsSymbol q_key,
+    DynamicsSymbol wTp_key, DynamicsSymbol wTc_key,
     const gtsam::noiseModel::Base::shared_ptr &cost_model,
     JointConstSharedPtr joint) {
   return boost::make_shared<gtsam::ExpressionFactor<gtsam::Vector6>>(
       cost_model, gtsam::Vector6::Zero(),
-      joint->poseConstraint(wTp_key.time()));
+      joint->poseConstraint(wTp_key.time(), wTp_key, wTc_key));
 }
```

Comment:
like `joint::poseConstraint`, maybe this constructor should be overloaded instead, so one constructor is like the previous version, and the other constructor is like

```c++
inline gtsam::NoiseModelFactor::shared_ptr PoseFactor(
    uint64_t time,
    const gtsam::noiseModel::Base::shared_ptr &cost_model,
    JointConstSharedPtr joint) {
  ...
}
```

Reply:
This constructor already exists and is the "canonical" one. The one I updated in this PR is the overloaded one.

Maybe I am not understanding something about your comment?

Reply:
oh ya my bad, I didn't see that